scx.full: 1.0.8 -> 1.0.9

JohnRTitor opened the pull request:
Changelog: https://github.com/sched-ext/scx/releases/tag/v1.0.9

Tested rustland, lavd, and central.

`aarch64-linux` build fails at the moment due to https://github.com/sched-ext/scx/issues/1320

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

JohnRTitor committed:
scx.full: 1.0.8 -> 1.0.9

Changelog: https://github.com/sched-ext/scx/releases/tag/v1.0.9

Signed-off-by: John Titor <50095635+JohnRTitor@users.noreply.github.com>

alyssais commented:
> `aarch64-linux` build fails at the moment due to [sched-ext/scx#1320](https://github.com/sched-ext/scx/issues/1320)

Should it be marked broken then?

JohnRTitor commented:
There's already a patch https://github.com/sched-ext/scx/issues/1320#issuecomment-2649640410

I'd rather have this merged as it is, then cherry pick the upstream patch from a PR/commit.

Gliczy commented:
LGTM on the `x86_64-linux` side.

But, is this update important enough to break `aarch64`? Wouldn't it be better to wait for a patch?

Gliczy commented:
I was assuming that there was some form of upstream support for `aarch64-linux`. 

In that case, it might be better to mark it as `broken` for the time being and maybe reconsider this in the future if upstream support for `aarch64-linux` improves and `scx` builds reliably in new releases.

Considering how relatively new `scx` is, it shouldn't affect too much people and it's probably better to be warned about `scx` being broken on `aarch64-linux` rather to have it randomly break when updating NixOS. And as you said, there is also the resource saving from Hydra if `scx` is broken.

JohnRTitor committed:
scx.full: don't build on aarch64-linux

As per https://github.com/NixOS/nixpkgs/pull/381138#issuecomment-2653368487
scx releases often breaks building on aarch64-linux, and besides the package
does not seem to have proper support for this platform

This reverts commit 0f40303097696a5cbb31cc583ae37c79540aa80f.

This can be reverted again if we feel upstream has improved its aarch64-linux support.
Link: https://github.com/sched-ext/scx/issues/1320

JohnRTitor commented:
I added to badPlatforms again.
